bgpd/bgp_label.c

```diff
@@ -120,6 +120,61 @@ mpls_label_t bgp_adv_label(struct bgp_dest *dest, struct bgp_path_info *pi,
 	return dest->local_label;
 }
 
+static void bgp_send_fec_register_label_msg(struct bgp_dest *dest, bool reg,
+					    uint32_t label_index)
+{
+	struct stream *s;
+	int command;
+	const struct prefix *p;
+	uint16_t flags = 0;
+	size_t flags_pos = 0;
+	mpls_label_t *local_label = &(dest->local_label);
+	bool have_label_to_reg =
+		bgp_is_valid_label(local_label)
+		&& label_pton(local_label) != MPLS_LABEL_IMPLICIT_NULL;
+
+	p = bgp_dest_get_prefix(dest);
+
+	/* Check socket. */
+	if (!zclient || zclient->sock < 0)
+		return;
+
+	/* If the route node has a local_label assigned or the
+	 * path node has an MPLS SR label index allowing zebra to
+	 * derive the label, proceed with registration. */
+	s = zclient->obuf;
+	stream_reset(s);
+	command = (reg) ? ZEBRA_FEC_REGISTER : ZEBRA_FEC_UNREGISTER;
+	zclient_create_header(s, command, VRF_DEFAULT);
+	flags_pos = stream_get_endp(s); /* save position of 'flags' */
+	stream_putw(s, flags);		/* initial flags */
+	stream_putw(s, PREFIX_FAMILY(p));
+	stream_put_prefix(s, p);
+	if (reg) {
+		/* label index takes precedence over auto-assigned label. */
```

Comment:
Please, see if debug log can be added under this section.

Reply:
done.